pkg/rdt/rdt.go

```diff
@@ -140,52 +139,65 @@ func SetLogger(l Logger) {
 }
 
 // Initialize discovers RDT support and initializes the  rdtControl singleton interface
-// NOTE: should only be called once in order to avoid adding multiple notifiers
-// TODO: support make multiple initializations, allowing e.g. "hot-plug" when
-// 		 resctrl filesystem is mounted
-func Initialize(resctrlGroupPrefix string, conf *Config) error {
+func Initialize(resctrlGroupPrefix string) error {
 	var err error
 
-	rdt = &control{Logger: log, resctrlGroupPrefix: resctrlGroupPrefix}
+	info = nil
+	rdt = nil
 
 	// Get info from the resctrl filesystem
-	rdt.info, err = getRdtInfo()
+	info, err = getRdtInfo()
 	if err != nil {
 		return err
 	}
 
-	// Configure resctrl
-	if err = rdt.setConfig(conf); err != nil {
-		return rdtError("configuration failed: %v", err)
-	}
+	r := &control{Logger: log, resctrlGroupPrefix: resctrlGroupPrefix}
+
+
+	rdt = r
```

Comment:
To address/enforce the 'should only be called once' restriction mentioned in the comment, maybe we shoud
  - introduce a `getRdtControl()` similar to the existing `getRdtInfo()`, and
  - turn both of them into `sync.Once.Do()`- or mutex-protected code blocks that initialize the singletons

`sync.Once.Do()` is probably not a good choice if we assume that the caller can/want to take corrective
actions for failed initialization and retry again, in which case mutex-protected singleton getters are needed.

Also, I should have mentioned this for the initial PR introducing `SetLogger()` but I forgot: I think with the current implementation of `log` being part of `control{}` we should give a warning or error if `SetLogger()` is called when `rdt != nil`, IOW, if it's already too late. Another option would be to make log an unexported global (and require it to be safe to be called by multiple goroutines).

Reply:
> To address/enforce the 'should only be called once' restriction mentioned in the comment,

If you're talking about `Initialize()`, this PR actually makes away with the restriction and it can be called multiple times.



> Also, I should have mentioned this for the initial PR introducing `SetLogger()` but I forgot: I think with the current implementation of `log` being part of `control{}` we should give a warning or error if `SetLogger()` is called when `rdt != nil`, IOW, if it's already too late

Very valid comment. I actually thought about this when implementing `SetLogger` but obviously forgot about that, too 🤔  I added a new patch addressing this. WDYT about that?

Reply:
Yes, I think that should be fine.

About the singletons, what is the reason of subsequent calls to `Initialize()` resetting and then re-creating `info` and `rdt` instead of returning already created ones ? Is the idea to explicitly support rediscovering potential (external) changes to resctrlfs (remounted with new options, or perhaps to a new path ?) by calling `Initialize()` again ?

Reply:
> Is the idea to explicitly support rediscovering potential (external) changes to resctrlfs (remounted with new options, or perhaps to a new path ?) by calling `Initialize()` again ?

Yes, that is the reasoning behind. Do full discovery/initialization on `Initialize()` as the mount might have changed/disappeared/(re-)appeared.

Reply:
Ok, then it looks fine to me.